Magento2/Sniffs/Commenting/ClassPropertyPHPDocFormattingSniff.php

```diff
@@ -125,23 +126,28 @@ public function processMemberVar(File $phpcsFile, $stackPtr)
             null,
             false
         );
-        if ($this->PHPDocFormattingValidator->providesMeaning(
-            $isShortDescriptionPreviousVar,
-            $commentStart,
-            $tokens
-        ) !== true) {
-            preg_match(
-                '`^((?:\|?(?:array\([^\)]*\)|[\\\\\[\]]+))*)( .*)?`i',
-                $tokens[($foundVar + 2)]['content'],
-                $varParts
-            );
-            if ($varParts[1]) {
-                return;
-            }
+
+        if (stripos($tokens[$isShortDescriptionPreviousVar]['content'], $tokens[$string]['content']) !== false) {
             $error = 'Short description duplicates class property name.';
-            $phpcsFile->addWarning($error, $isShortDescriptionPreviousVar, 'AlreadyHaveMeaningFulNameVar');
+            $phpcsFile->addWarning($error, $isShortDescriptionPreviousVar, 'AlreadyHaveMeaningfulNameVar');
             return;
         }
+        $re = '/
+                # Split camelCase "words". Two global alternatives. Either g1of2:
+                  (?<=[a-z])      # Position is after a lowercase,
+                  (?=[A-Z])       # and before an uppercase letter.
+                | (?<=[A-Z])      # Or g2of2; Position is after uppercase,
+                  (?=[A-Z][a-z])  # and before upper-then-lower case.
+                /x';
+        $varTagParts = preg_split($re, $tokens[$string]['content']);
+
+        foreach ($varTagParts as $part) {
+            if (stripos($tokens[$isShortDescriptionPreviousVar]['content'], $part) === false) {
+                return;
+            }
+        }
```

Comment:
I don't think finding a part of property name in the description should fail the test, I'd rather check if the short description matches the split variable name.
```suggestion
            if (stripos($tokens[$isShortDescriptionPreviousVar]['content'], implode(' ', $varTagParts)) === false) {
                return;
            }
```